runtime/compiler/control/CompilationThread.cpp

```diff
@@ -2993,15 +2993,17 @@ void TR::CompilationInfo::insertDLTRecord(J9Method *method, int32_t bcIndex, voi
    }
    }
 
-void TR::CompilationInfo::cleanDLTRecordOnUnload(J9ClassLoader *classloader)
+void TR::CompilationInfo::cleanDLTRecordOnUnload()
    {
    for (int32_t i=0; i<DLT_HASHSIZE; i++)
       {
       struct DLT_record *prev=NULL, *curr=_dltHash[i], *next;
       while (curr != NULL)
          {
+         J9Class *clazz = J9_CLASS_FROM_METHOD(curr->_method);
          next = curr->_next;
-         if (J9_CLASS_FROM_METHOD(curr->_method)->classLoader == classloader)
+         if ( J9_ARE_ALL_BITS_SET(clazz->classLoader->gcFlags, 0x2)
```

Comment:
Don't hardcode `0x2` here as we'll miss it if we every re-order the flags.   There should be a named constant that can be used instead

Reply:
We could either, 

Move the original defines on the GC side to somewhere central.
https://github.com/eclipse/openj9/blob/29cc3dba911e30df27e2856c50f6b9ea78508fe3/runtime/gc_include/j9modron.h#L179-L185

Or, we could just define `J9_GC_CLASS_LOADER_DEAD` in CompilationThread.

Reply:
Why not include j9modron.h directly?

Reply:
My apologies, I thought I tried earlier and got some re-definition errors. 
Included J9modron.h directly. See https://github.com/AlenBadel/openj9/commit/5095b2f287b4a7a6dcbd248634772daaa299a8f5